pkg/ebpf/c/tracee.bpf.c

```diff
@@ -6289,3 +6313,836 @@ int tc_ingress(struct __sk_buff *skb)
+    switch (protocol) {
+        // case IPPROTO_IPV6:
+        // case IPPROTO_IPIP:
+        // case IPPROTO_DCCP:
+        // case IPPROTO_SCTP:
+        // case IPPROTO_UDPLITE:
+        case IPPROTO_IP:
+        case IPPROTO_TCP:
+        case IPPROTO_UDP:
+        case IPPROTO_ICMP:
+        case IPPROTO_ICMPV6:
+            break;
+        default:
+            return 0;
+    }
+
+    return 1;
+    // clang-format on
+}
+
+//
+// network related maps
+//
+
+// clang-format off
+
+// cgroupctxmap
+
+#define SUB_NET_PACKET_IP      (1 << 0) // submit net_packet_ip_base
+#define SUB_NET_PACKET_TCP     (1 << 1) // submit net_packet_tcp_base
+#define SUB_NET_PACKET_UDP     (1 << 2) // submit net_packet_udp_base
+#define SUB_NET_PACKET_ICMP    (1 << 3) // submit net_packet_icmp_base
+#define SUB_NET_PACKET_ICMPV6  (1 << 4) // submit net_packet_icmpv6_base
+#define SUB_NET_PACKET_DNS     (1 << 5) // submit net_packet_dns_base
+
+typedef struct net_event_context {
+    event_context_t eventctx;
+    struct { // event arguments (needs packing), use anonymous struct to ...
+        u8 index0;
+        u32 bytes;
+        // ... (payload sent by bpf_perf_event_output)
+    } __attribute__((__packed__)); // ... avoid address-of-packed-member warns
+    // members bellow this point are metadata (not part of event to be sent)
+    u8 should_submit;
+    u32 header_size;
+} net_event_context_t;
+
+struct {
+    __uint(type, BPF_MAP_TYPE_HASH);
+    __uint(max_entries, 1024);          // simultaneous cgroup/skb ingress/eggress progs
+    __type(key, u64);                   // sk_buff timestamp
+    __type(value, net_event_context_t); // event context built so cgroup/skb can use
+} cgrpctxmap SEC(".maps");              // saved info between SKB caller and SKB program
+
+// inodemap
+
+typedef struct net_task_context {
+    struct task_struct *task;
+    task_context_t taskctx;
+} net_task_context_t;
+
+struct {
+    __uint(type, BPF_MAP_TYPE_LRU_HASH);
+    __uint(max_entries, 10240);             // simultaneous sockets being traced
+    __type(key, u64);                       // socket inode number ...
+    __type(value, struct net_task_context); // ... linked to a task context
+} inodemap SEC(".maps");                    // relate sockets and tasks
+
+// entrymap
+
+typedef struct entry {
+    long unsigned int args[6];
+} entry_t;
+
+struct {
+    __uint(type, BPF_MAP_TYPE_HASH);
+    __uint(max_entries, 1024);       // simultaneous tasks being traced for entry/exit
+    __type(key, u32);                // host thread group id (tgid or tid) ...
+    __type(value, struct entry);     // ... linked to entry ctx->args
+} entrymap SEC(".maps");             // can't use args_map (indexed by existing events only)
+
+// clang-format on
+
+//
+// support functions for network code
+//
+
+static __always_inline u64 sizeof_net_event_context_t(void)
+{
+    return sizeof(net_event_context_t) - 7; // neteventctx metadata & padding
```

Comment:
Yep.. it will brake on practically any change of "net_event_context_t" (I have been there, as I started the code having 3 arguments there just to test things out and the value changed each time an argument is added/removed). Let's see...

Reply:
Alright.. made it "automatic" by doing: 

```C
typedef struct net_event_contextmd {
    u8 should_submit;
    u32 header_size;
    u16 padding;
} __attribute__((__packed__)) net_event_contextmd_t;

typedef struct net_event_context {
    event_context_t eventctx;
    struct { // event arguments (needs packing), use anonymous struct to ...
        u8 index0;
        u32 bytes;
        // ... (payload sent by bpf_perf_event_output)
    } __attribute__((__packed__)); // ... avoid address-of-packed-member warns
    // members bellow this point are metadata (not part of event to be sent)
    net_event_contextmd_t md;
} net_event_context_t;
```

Reply:
Then I can use:
```C
static __always_inline u64 sizeof_net_event_context_t(void)
{
    return sizeof(net_event_context_t) - sizeof(net_event_contextmd_t);
}
```